storage/src/vespa/storage/common/content_bucket_space_repo.h

```diff
@@ -0,0 +1,24 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+#include "content_bucket_space.h"
+#include <vespa/document/bucket/bucketspace.h>
+#include <unordered_map>
+
+namespace storage {
+
+/**
+ * Class managing the set of bucket spaces (with associated bucket databases) on a content node.
+ */
+class ContentBucketSpaceRepo {
+private:
+    using BucketSpaceMap = std::unordered_map<document::BucketSpace, ContentBucketSpace::UP, document::BucketSpace::hash>;
+
+    BucketSpaceMap _map;
+
+public:
+    ContentBucketSpaceRepo();
+    ContentBucketSpace *get(document::BucketSpace bucketSpace) const;
```

Comment:
Do we want the future semantics of the function (I'm saying future, as it currently asserts validity) to be that an unknown bucket space returns `nullptr`, or shall the returned pointer always be valid?. In case of the latter, returning a reference would make it clear that the caller does not need to perform a null-check on the returned value.